### PR TITLE
Adds IDs and Wallets

### PR DESCRIPTION
**Pull request by Arokha**
Adds variants of ID cards for each department and head, and adds several new colors of wallet. The wallets self-color the little ID in the corner based on the ID inserted so they always show the right picture of the ID, without needing to sprite each ID. Some are snowflaked, notably Gold and Centcom, but the rest are generated on the fly based on colors on the ID.

Per the forum thread here:
http://ss13polaris.com/forums/viewtopic.php?f=18&t=1004&p=9572

new IDs

(White wallet not depicted)

### CODE REVIEW

**Comment by Arokha** [comment]
(White wallet not depicted)

**Comment by Yoshax** [comment]
Which is which in the screenshot?

**Comment by Arokha** [comment]
Each row is the same department, showing head (gold bars) and non-head, from the top:

- Captain (by itself)
- HoP and civilian (bartender, chef, etc)
- Medical
- Security
- Engineering
- Science
- Cargo

Also in the cards.dmi file there are head and non-head versions of some unused colors, like green, cyan, pink, and several unused ones like extra syndie IDs.

**Comment by SpadesNeil** [comment]
Gold trim is head staff!

File: code/game/objects/items/weapons/storage/wallets.dm [hunk 1]
```diff
@@ -45,23 +46,59 @@
...
+				del id_overlay
+				icon = new_wallet
+			else //Doesn't match a special overlay type.
+				if(front_id.primary_color && front_id.secondary_color) //Colored ID with stripe and oval colors (pri/sec).
+					var/icon/new_wallet = new/icon("icon" = initial(icon), "icon_state" = initial(icon_state))
+					var/icon/id_icon = new/icon("icon" = initial(icon), "icon_state" = "wallet_id")
+					var/icon/pri_overlay = new/icon("icon" = initial(icon), "icon_state" = "wallet_idprimary")
+					var/icon/sec_overlay = new/icon("icon" = initial(icon), "icon_state" = "wallet_idsecondary")
+
+					pri_overlay.Blend(front_id.primary_color, ICON_ADD)
```
Comment (mwerezak) [inline code/game/objects/items/weapons/storage/wallets.dm]: Icon procs...

**Comment by Datraen** [comment]
I'd much rather see actual wallet icons for each of the departments, rather than having BYOND manipulate icons.

**Comment by Arokha** [comment]
Erm. That'd require 91 wallet sprites, for JUST those IDs in the picture (plus white wallet which I forgot to include). It'd probably require somewhere around 160 if you did all the ID sprites for syndie and whatnot.

**Comment by Datraen** [comment]
To clarify: Instead of having ID icons generated as you go (That is, coloring a basic ID overlay), I'd rather see the IDs themselves have their own overlay icons for the wallet.

I'm not asking for each individual wallet combination to be sprited, that would be completely asinine and unreasonable.

**Comment by Arokha** [comment]
> I'm not asking for each individual wallet combination to be sprited, that would be completely asinine and unreasonable.

That is how it was originally, so I thought you meant going back to that.

**Comment by Datraen** [comment]
Leaving the fall-back system in place would be good to not break anything, but it shouldn't be the preferred method of handling the sprites.

**Comment by Arokha** [comment]
Well, you can't really check for the existence of iconstates, so unfortunately the code would not know when to fall back. In this code, it falls back if the ID doesn't have a defined pri/sec color.

**Comment by PsiOmegaDelta** [comment]
```if(”my_icon_state" in icon_states(icon))```

**Comment by Arokha** [comment]
Yeh, just discovered that. Used it to generate the overlays to save me from spriting them. I'll commit an updated version of this in a minute with all the tiny generated ID icons.

**Comment by mwerezak** [comment]
Also you should be aware that icon procs are super slow and you should try to use images wherever possible.

**Comment by Arokha** [comment]
I'm gonna close this because I pretty much rewrote how this works and I don't really feel like a PR should be the same PR if the description is entirely different, and if all the comments are not applicable anymore.

**Pull request closed without merging**